### PR TITLE
expat: Update to 2.6.4

### PR DESCRIPTION
**Pull request by mtl1979**
* Old versions have been renamed in SourceForge downloads to discourage people from downloading vulnerable versions, so recipes are broken.

### CODE REVIEW

**Comment by Begasus** [comment]
SONAME changed, checked if it doesn't brake other packages requiring this?

**Comment by mtl1979** [comment]
> ~SONAME~ libVersion changed, checked if it doesn't brake other packages requiring this?

Both versions should have libexpat.so.1 as SONAME, so nothing should break. If something breaks, it can't be worse than recipe(s) that don't build at all...

**Comment by korli** [comment]
Please restore 2.5.0 for x86_gcc2. Old versions at Sourceforge have been renamed, not removed.

**Comment by mtl1979** [comment]
> Please restore 2.5.0 for x86_gcc2. Old versions at Sourceforge have been renamed, not removed.

They were "renamed" so people wouldn't use the vulnerable versions... For now, I fixed the download link for 2.5.0 ... I can't guarantee how long the "fixed" link will work...

File: dev-libs/expat/expat-2.5.0.recipe
Comment (jmairboeck) [inline dev-libs/expat/expat-2.5.0.recipe:1]: I think ARCHITECTURES of this should be changed to just "x86_gcc2" instead of "all" and remove SECONDARY_ARCHITECTURES. (I can't comment on the lines directly because they aren't part of the diff.)

File: dev-libs/expat/expat-2.5.0.recipe
Comment (Begasus) [inline dev-libs/expat/expat-2.5.0.recipe:9]: Changing the URI should be enough here if there is no real reason to rebuild it, this would keep the legacy version currently in the depot.

File: dev-libs/expat/expat-2.5.0.recipe
Comment (korli) [reply dev-libs/expat/expat-2.5.0.recipe:1]: What do you insist on fixing the recipe? The package already exists in the repo.

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: What you are saying makes absolutely no sense. Someone who has worked on Haiku as long as you have, should already know the answer. We don't fix the recipe for HaikuDepot, we fix them for users who want to build things locally using latest available beta 5 patch release.

Comment (korli) [reply dev-libs/expat/expat-2.5.0.recipe:1]: I don't get what you mean. Who wants to build things locally again?

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: The current binaries are built against earlier patch release, which means "haiku" and "haiku_devel" packages for that patch release must be available on the machine... package_daemon does not allow having unlimited number of versions of single package. It's possible currently to have two versions without package_daemon committing seppuku during package upgrade.

Comment (korli) [reply dev-libs/expat/expat-2.5.0.recipe:1]: expat-2.5.0 for gcc2 was built on r1beta4, and should be compatible with r1beta5 and nightly builds. Isn't it the case?

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: Even packages built on beta5+113 (beta 5 install ISO) aren't compatible with beta5+125 (which is latest patch release)...

Comment (Begasus) [reply dev-libs/expat/expat-2.5.0.recipe:1]: So everything build on buildmasters using beta5+111 aren't compatible? I'm not a developer, and not as much occupied with 32bit as I used to be, but so far I haven't seen an issue/problem reported for our current expat version for gcc2?

Comment (Begasus) [reply dev-libs/expat/expat-2.5.0.recipe:1]: OK way to far over my head, I only know I never had any real problems with this package, and I've been around for quite some time :)
I'll let the masters continue on this.

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: I had no issues until beta5+125, because before that there was only ever 2 patch releases used for packaging binaries. But this PR is not about bugs in or fixing package_daemon, this PR is about updating the expat recipe(s) to non-vulnerable version... So far no-one has explained why we specifically need version 2.5.0 for gcc2 or why the newer versions don't work or shouldn't be used for gcc2.

Until the questions have been adequately answered, only thing we should in my opinion do in the recipe for version 2.5.0 is to keep the download link working. The rest has always been open and free to discussion.

Comment (korli) [reply dev-libs/expat/expat-2.5.0.recipe:1]: When building yourself with Haikuporter, you should full-sync on the stable branch, empty the packages folder at the haikuports root. Otherwise you'll be on your own.

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: I had fresh 32-bit Haiku installation, so everything was up-to-date with "pkgman full-sync"... Obviously after the recipe update, some packages are incompatible with the prebuilt binary packages on official repositories. This should fix itself after this PR is merged and repository is updated.

Comment (korli) [reply dev-libs/expat/expat-2.5.0.recipe:1]: Nonetheless no one is seeing the same issue AFAICT

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: I was only rebuilding expat on clean installation running +125, but in the end there was 148 packages in haikuports/packages, because all the packages that use expat and their dependencies had to be rebuilt as HaikuPorter does require exact same version of the each package as was originally used. The binary packages that are currently available on beta 5 are definitely not the same ones that were used to build those (binary) packages. Some of the packages couldn't even be built as the recipes were broken partly due to builders not using strict mode, which is currently the default, and partly due to people merging recipes that fail to build due to features not implemented in current Haiku releases (beta 5 or nightly). I've talked with @Begasus on Matrix and IRC about trying to get most of the broken packages fixed, but some packages are beyond all hope and some need fixes in packages or libraries not available on HaikuPorts, but instead are built along with Haiku sources.

As this is repository for HaikuPorts and not Haiku itself, this limitation of HaikuPorter is quite relevant.

Comment (Begasus) [reply dev-libs/expat/expat-2.5.0.recipe:1]: The buildmasters not using strict mode is a known thing and have been for as long as I remember, even had discussions about it back then, there is a reason for it.
I've checked a few gcc2 recipes using libexpat and had no problems building them, freetype_bootstrap is one that needs strict mode to be off, for obvious reason, aside from that it builds perfectly fine with expat_x86, as mentioned on IRC.
I know for one that we do as much as we can to avoid merging packages that are known to fail, and if so comment on the failed recipes after buildmasters failure. 
I still don't see a reason why expat should be rebuild for gcc2.

Comment (mtl1979) [reply dev-libs/expat/expat-2.5.0.recipe:1]: I thought not bumping the revision meant that expat won't be rebuilt for gcc2... If that isn't the case, then I can drop the change for 2.5.0 recipe from this PR.

What comes to not having issues on other installations, I think the main difference is what HaikuPorter thinks is the base version of "haiku_devel" package and if that package is actually available on the build host... As far as I have understoon, HaikuPorter will decide the base version for "haiku_devel" on first run, so it can be +113, +122 or +125 in most cases for fresh installations of beta 5 depending on if pkgman or SoftwareUpdater was run before HaikuPorter or not.

**Review by Begasus — APPROVED** [review]
Still need to check on a running system, but LGTM.

**Review by Begasus — APPROVED** [review]

**Pull request merged**
